Pokedex Challenge/Pokedex Challenge/Views/View Controllers/PokemonDetailViewController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  PokemonDetailViewController.swift
+//  Pokedex Challenge
+//
+//  Created by Michael Flowers on 1/25/19.
+//  Copyright © 2019 Michael Flowers. All rights reserved.
+//
+
+import UIKit
+
+class PokemonDetailViewController: UIViewController, UISearchBarDelegate {
+
+    //MARK: - Properties
+    var pokemon: Pokemon? {
+        didSet {
+            updateViews()
+        }
+    }
+    var pokemonController: PokemonController?
+    
+    //MARK: - IBOutlets
+    @IBOutlet weak var searchBar: UISearchBar!
+    @IBOutlet weak var pokemonName: UILabel!
+    @IBOutlet weak var id: UILabel!
+    @IBOutlet weak var types: UILabel!
+    @IBOutlet weak var abilities: UILabel!
+    @IBOutlet weak var saveButtonProperties: UIButton!
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        //set the searchBar delegate
+       searchBar.delegate = self
+    }
+    
+    //when user taps cell, updateViews when the view will appear again.
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        updateViews()
+    }
+    
+    //MARK: - PokemonController functions
+    @IBAction func savePokemon(_ sender: UIButton) {
+        navigationController?.popViewController(animated: true)
+    }
+    
+    //search bar delegate function
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        //fetch the pokemon
+        
+        guard let name = searchBar.text, !name.isEmpty else { return }
+        pokemonController?.fetchPokemon(with: name, completion: { (error) in
+            if let error = error {
+                NSLog("Error fetching the pokemon from searchBar: \(error.localizedDescription)")
+                return
+            }
+            //this is where we append it to the array
```

Comment:
No we don't append it to the array yet. We need to check pokemon and present it to the user. And also fetch it's image
```suggestion
            self.pokemon = pokemon
            self.pokemonController?.fetchImage(of: pokemon, completion: { (image) in
                guard let image = image else {
                    NSLog("No image returned")
                    return
                }
                
                DispatchQueue.main.async {
                    self.pokemonImageView.image = image
                }
            })
```